torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -466,6 +483,243 @@ def codegen_define(self, kernel: CppTemplateKernel) -> str:
         return result
 
 
+@register_micro_gemm(
+    *generate_gemm_config(
+        VecAVX512,
+        [(4, 64, 32), (4, 64, 64)],
+        input_dtype=torch.bfloat16,
+        input2_dtype=torch.int32,
+        output_dtype=torch.float,
+        compute_dtype=torch.float,
+    ),
+)
+class CppMicroGemmInt4WoQVec(CppMicroGemmFP32Vec):
+    """
+    This class generates the code for int4 WoQ micro gemm using AVX512 intrinsics.
+    It is based on the corresponding ATen kernel.
+    """
+
+    TEMPLATE_ENTRY = r"""
+{{declare_kernel}} {
+    TORCH_CHECK(N % {{block_n}} == 0, "N dimension must be multiple of {{block_n}}");
+    TORCH_CHECK(K % {{block_k}} == 0, "K dimension must be multiple of {{block_k}}");
+    // TODO(jgong5): loop unroll for M and N
+    for (int64_t m = 0; m < M; m += {{block_m}}) {
+        int64_t block_m = std::min<int64_t>(M - m, {{block_m}});
+        for (int64_t n = 0; n < N; n += {{block_n}}) {
+            if (block_m == {{block_m}}) {
+                {{kernel_name}}_kernel<{{block_m}}, {{block_n}}, accum>(
+                    A + m * lda,
+                    reinterpret_cast<const uint8_t*>(B) + n,
+                    ZPS,
+                    C + m * ldc + n,
+                    K,
+                    lda,
+                    ldb,
+                    ldc,
+                    actual_N,
+                    k_start,
+                    q_group_size
```

Comment:
@jgong5, the existing AVX512 micro-kernel template would not only need special-casing for adding 4 more input arguments, but would also need a conditional `reinterpret_cast<const uint8_t*>(B)` (unless I already do it in the GEMM template with a view).